packages/@lwc/rollup-plugin/src/index.ts

```diff
@@ -70,21 +70,31 @@ function isImplicitCssImport(importee: string, importer: string): boolean {
     );
 }
 
-interface scopedOption {
+interface Descriptor {
     filename: string;
     scoped: boolean;
+    specifier: string | null;
 }
 
-function parseQueryParamsForScopedOption(id: string): scopedOption {
+function parseDescriptorFromFilePath(id: string): Descriptor {
     const [filename, query] = id.split('?', 2);
-    const params = query && new URLSearchParams(query);
-    const scoped = !!(params && params.get('scoped'));
+    const params = new URLSearchParams(query);
+    const scoped = params.has('scoped');
+    const specifier = params.get('specifier');
     return {
         filename,
+        specifier,
         scoped,
     };
 }
 
+function appendAliasSpecifierQueryParam(id: string, specifier: string): string {
+    const [filename, query] = id.split('?', 2);
+    const params = new URLSearchParams(query);
+    params.set('specifier', specifier);
+    return `${filename}?${params.toString()}`;
+}
```

Comment:
Are we going to have to make this change in `lwc-platform`, `lwc-webpack-plugin`, `lwc-test`, and LWR as well? Usually when we add new query params, we have to.

Reply:
Looking through these repositories, it looks like `lwc-webpack-plugin` and `lwc-test` will need a similar update.

`lwc-platform` invokes the transformers directly and passes in the namespace / name that's provided as part of the bundle so there's no need for LWC to scan the file directory.

It looks like the LWR code just calls the rollup plugin directly, so they should be able to use the plugin as is.